### PR TITLE
Basic subsystem docs update

### PR DESCRIPTION
**Pull request by omri00**

### CODE REVIEW

**Comment by omri00** [comment]
need the docs for target and PID for speed

**Comment by omri00** [comment]
needs new docs for Xbox

**Comment by omri00** [comment]
also the numerical target

File: src/com/spikes2212/utils/PIDSettings.java [hunk 1]
```diff
@@ -41,127 +44,121 @@ public PIDSettings(double KP, double KI, double KD, double tolerance, double wai
...
 
 	/**
-	 * 
-	 * Sets the Differential coefficient of the PID loop using this settings.
+	 * Sets the Differential coefficient of the PID loop using these
+	 * {@link PIDSettings}.
 	 *
-	 * @param KD
+	 * @param KP
 	 *            the new Differential coefficient.
```
Comment (TomerSitton) [inline src/com/spikes2212/utils/PIDSettings.java]: it's KD

File: src/com/spikes2212/utils/PIDSettings.java [hunk 2]
```diff
@@ -41,127 +44,121 @@ public PIDSettings(double KP, double KI, double KD, double tolerance, double wai
...
 	public void setKD(double KD) {
 		this.KD = KD;
 	}
 
 	/**
-	 * Gets the time this PID loop will wait while within tolerance of the
-	 * setpoint before ending.
-	 * <p>
-	 * The PID control of the subsystem continues while waiting
-	 * </p>
```
Comment (TomerSitton) [inline src/com/spikes2212/utils/PIDSettings.java]: why did you remove this line? (The PID control of the subsystem continues while waiting)

File: src/com/spikes2212/utils/RunnableCommand.java [hunk 3]
```diff
@@ -3,42 +3,50 @@
...
+	 * 
+	 * <br>
+	 * <br>
+	 * This command finishes when the thread finishes, meaning when run stops.
+	 *
+	 * @param runnable
+	 *            the {@link Runnable} to run whenever this command starts
+	 * @param runInDisabled
+	 *            whether this command should be allowed to run when the robot is
+	 *            disabled. Default is true.
```
Comment (TomerSitton) [inline src/com/spikes2212/utils/RunnableCommand.java]: there is no such parameter

File: src/com/spikes2212/utils/VoltageMonitor.java [hunk 4]
```diff
@@ -4,11 +4,29 @@
 
 import edu.wpi.first.wpilibj.PowerDistributionPanel;
 
+/**
+ * This class is ment to decrease the posibility of power drops while running by
+ * decreasing the engine power according the the current voltage.<br>
+ * Voltage is messured in Volts.
```
Comment (TomerSitton) [inline src/com/spikes2212/utils/VoltageMonitor.java]: not engine, motors

File: src/com/spikes2212/utils/VoltageMonitor.java [hunk 5]
```diff
@@ -53,12 +75,18 @@ public static double getLowVoltage() {
...
+	 * @param lowVoltage
+	 *            the Voltage considered LOW
+	 * 
+	 * @return a supplier of values to set, changing according to the voltage on the
+	 *         system
+	 * 
+	 * @see Supplier
+	 */
     public static Supplier<Double> monitorSupplier(Supplier<Double> supplier,
                                                    double highVoltage, double lowVoltage) {
```
Comment (TomerSitton) [inline src/com/spikes2212/utils/VoltageMonitor.java]: where is the doc for this method? there are only the parameters of it

File: src/com/spikes2212/genericsubsystems/utils/limitationFunctions/TwoLimits.java [hunk 6]
```diff
@@ -6,11 +6,12 @@
 import com.spikes2212.genericsubsystems.BasicSubsystem;
 
 /**
- * This is a Function from double to boolean. This is for the constructor a
- * {@link BasicSubsystem} that moves between two limits.
+ * This is a {@link Function} from double to boolean. This is for the
+ * constructor a {@link BasicSubsystem} that moves between two given limits.
```
[the remark this answers is not in the document]
Comment (itamargreen) [reply src/com/spikes2212/genericsubsystems/utils/limitationFunctions/TwoLimits.java]: urrm. nope. (I'll delete this comment soon...)

`constructor `**`of`**` a basic`...

File: src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemWithPID.java [hunk 7]
```diff
@@ -108,31 +81,84 @@ public MoveBasicSubsystemWithPID(BasicSubsystem basicSubsystem, PIDSource source
...
+	
+	/**
+	 * Sets the time this command will wait while within tolerance of the setpoint
+	 * before ending.
+	 * 
+	 * <br>
+	 * <br>
+	 * The PID control of the subsystem continues while waiting. <br/>
+	 * If wait time is set to 0, the command will not wait after reaching the
+	 * setpoint.
```
[the remark this answers is not in the document]
Comment (villigernu7) [reply src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemWithPID.java]: which one?

File: src/com/spikes2212/genericsubsystems/BasicSubsystem.java [hunk 8]
```diff
@@ -69,10 +74,8 @@ public double getSpeed() {
 	}
 
 	/**
-	 * Sets the default command. If this is not called or is called with null,
-	 * then there will be no default command for the subsystem.
-	 * 
-	 * @see edu.wpi.first.wpilibj.command.Subsystem#setDefaultCommand(edu.wpi.first.wpilibj.command.Command)
+	 * Sets the default command. If this is not called, or is called with null, then
+	 * there will be no default command for the subsystem.
```
[the remark this answers is not in the document]
Comment (TomerSitton) [reply src/com/spikes2212/genericsubsystems/BasicSubsystem.java]: you need to use see for the default command on wpi.

File: src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemWithPID.java [hunk 9]
```diff
@@ -34,37 +34,37 @@
...
-	 * The target distance. Units according to {@link #source}.
-	 */
 	protected final PIDSettings PIDSettings;
-	
+	protected final Supplier<Double> setpoint;
 	/**
 	 * The <a href=
 	 * "http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/PIDSource.html">PIDSource<a>
 	 * this subsystem uses, given by {@link BasicSubsystem#getPIDSource()}.
 	 */
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/commands/MoveBasicSubsystemWithPID.java]: remove this doc

File: src/com/spikes2212/genericsubsystems/utils/limitationFunctions/Limitless.java [hunk 10]
```diff
@@ -5,7 +5,7 @@
 import com.spikes2212.genericsubsystems.BasicSubsystem;
 
 /**
- * This is a {@link Function} that always returns true. This is
+ * This is a {@link Function} from Double to Bollean that always returns true. This is
  * for the constructor of a {@link BasicSubsystem} with no limits.
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/utils/limitationFunctions/Limitless.java]: bollean?

Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/utils/limitationFunctions/Limitless.java]: u can write something like "an instance of this class can used as the canMove condition in the constructor..."

File: src/com/spikes2212/genericsubsystems/utils/limitationFunctions/TwoLimits.java [hunk 11]
```diff
@@ -6,8 +6,8 @@
 import com.spikes2212.genericsubsystems.BasicSubsystem;
 
 /**
- * This is a {@link Function} from double to boolean. This is for the
- * constructor a {@link BasicSubsystem} that moves between two given limits.
+ * This is a {@link Function} from Double to Boolean. This is for the
+ * constructor of a {@link BasicSubsystem} that moves between two given limits.
  *
```
Comment (TomerSitton) [inline src/com/spikes2212/genericsubsystems/utils/limitationFunctions/TwoLimits.java]: see comment in limitless
